libraries/WiFi/src/WiFiClient.cpp

```diff
@@ -289,7 +289,7 @@ int WiFiClient::getOption(int option, int *value)
     size_t size = sizeof(int);
     int res = getsockopt(fd(), IPPROTO_TCP, option, (char *)value, &size);
     if(res < 0) {
-        log_e("%d", errno);
+        log_e("getOption fail on fd %d, errno: %d", fd(), errno);
```

Comment:
the name of the function is not necessary. the log facility will print the file name, function name and line number.
AFAIK IDF has a facility to turn the errorno into string :) check the sdk folder. Path should make sense

Reply:
that's cool... I was just following through on comments from @atanisoft... looks like esp_err_to_name is the function you are thinking of?

Reply:
addressed by 3e7dd27e0e937203b963c1af3c8eba3f6b267650

Reply:
addressed by 3e7dd27e0e937203b963c1af3c8eba3f6b267650

Reply:
uhmm not sure if this will work for errno though. The method you posted is for esp_err_t

Reply:
@me-no-dev @vicatcu it won't work for errno, the proper function is strerror(errno)

Reply:
Thanks @atanisoft, addressed by https://github.com/espressif/arduino-esp32/pull/2663